packages/radix-ui-themes/src/helpers/get-root.tsx

```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
+import { Slot } from '@radix-ui/react-slot';
+
+interface GetChildrenArgs<T extends React.ElementType> {
+  asChild: boolean | undefined;
+  children: React.ReactNode;
+  parent: T;
+}
+
+export const getRoot = <
+  T extends React.ElementType,
+  U extends React.ComponentProps<T> extends { asChild?: boolean } ? T : Exclude<T, typeof Slot>
+>({
+  asChild,
+  children,
+  parent: Parent,
+}: GetChildrenArgs<T>): {
+  Root: U;
+  children: React.ReactNode;
+} => {
+  if (asChild) {
+    let child = React.Children.only(children) as React.ReactElement;
+    const grandChildren = child.props.children;
+    return {
+      Root: ((props) => {
+        child = React.cloneElement(child, {
+          children: props.children,
+        });
+
+        // Make sure we don't pass `asChild` to DOM elements
+        const asChildProp = (Parent as unknown) === Slot ? undefined : asChild;
```

Comment:
@benoitgrelard I did see React complain about an unrecognized attribute

Reply:
I still see one on the sink page but might be coming from somewhere?

Reply:
Found it, was extracting it from the wrong place
2c6eb447b6e4d166fb8aeb7d45102da42ca46507